### PR TITLE
Ran/active panel screen share

### PR DESCRIPTION
**Pull request by stormcrow85**

### CODE REVIEW

File: src/components/activePanel/svg/track-teacher-icon.svg [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Generator: Adobe Illustrator 19.0.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
+<svg version="1.1" id="Isolation_Mode" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px"
+	 y="0px" viewBox="-326 51.7 138 141.3" style="enable-background:new -326 51.7 138 141.3;" xml:space="preserve">
+<style type="text/css">
+	.st0{fill:none;stroke:#000000;stroke-width:6;stroke-linecap:round;stroke-linejoin:round;stroke-miterlimit:10;}
```
Comment (igormas) [inline src/components/activePanel/svg/track-teacher-icon.svg]: ?

File: src/components/activePanel/enums/activePanelStatusEnum.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+(function (angular) {
+    'use strict';
+
+    angular.module('znk.infra.activePanel').factory('ActivePanelStatusEnum',
+        function (EnumSrv) {
+            'ngInject';
+
+            return new EnumSrv.BaseEnum([
```
Comment (igormas) [inline src/components/activePanel/enums/activePanelStatusEnum.js]: Do we really need it as a global enum ?

File: src/components/activePanel/enums/activePanelComponentEnum.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+(function (angular) {
+    'use strict';
+
+    angular.module('znk.infra.activePanel').factory('ActivePanelComponentEnum',
+        function (EnumSrv) {
+            'ngInject';
+
+            return new EnumSrv.BaseEnum([
```
Comment (igormas) [inline src/components/activePanel/enums/activePanelComponentEnum.js]: Do we really need it as a global enum ?

File: src/components/activePanel/activePanel.service.js [hunk 4]
```diff
@@ -2,25 +2,110 @@
...
+            function _base(name, param1) {
                 var fn = actions[name];
                 if (angular.isFunction(fn)) {
-                    fn(arg1);
+                    fn(param1);
                 }
             }
 
-            this.showActivePanelDrv = _base.bind(null, 'showUI');
+            var showActivePanelDrv = _base.bind(null, 'showUI');
```
Comment (igormas) [inline src/components/activePanel/activePanel.service.js]: Good thinking, better implementation will be iterating over the desired actions.

File: src/components/activePanel/activePanel.service.js [hunk 5]
```diff
@@ -2,25 +2,110 @@
...
+                function isScreenSharingActive() {
+                    return (currentStatus.screenSharing === ActivePanelStatusEnum.ACTIVE.enum);
+                }
+
+                function isCallActive() {
+                    return (currentStatus.calls === ActivePanelStatusEnum.ACTIVE.enum);
+                }
+
+                // default for show drv = false
+                switch (true) {
```
Comment (igormas) [inline src/components/activePanel/activePanel.service.js]: ?

File: src/components/activePanel/activePanel.service.js (hunk 4 above)
Comment (stormcrow85) [inline src/components/activePanel/activePanel.service.js]: explain :)

File: src/components/activePanel/activePanel.directive.js [hunk 6]
```diff
@@ -11,38 +11,56 @@
...
-                    }
+                    };
+
+                    scope.actions.stopTimer = function () {
+                        $interval.cancel(timerInterval);
+                    };
+
+                    scope.actions.screenShareMode = function (bool) {
+                        $log.debug('screenShareMode');
+                        if (bool) {
```
Comment (igormas) [inline src/components/activePanel/activePanel.directive.js]: it better to name it 'isScreenShareMode' rather then bool

**Pull request merged**